app/src/main/java/io/reark/rxgithubapp/advanced/data/DataLayer.java

```diff
@@ -38,13 +38,14 @@
 import io.reark.rxgithubapp.shared.network.GitHubService;
 
 import static io.reark.reark.utils.Preconditions.checkNotNull;
-import static io.reark.reark.utils.Preconditions.get;
 
 public class DataLayer extends ClientDataLayerBase {
     private static final String TAG = DataLayer.class.getSimpleName();
 
     private final Context context;
 
+    private int nextListenerIndex = 0;
```

Comment:
Good point with the processes getting the same listener id, I'll make it a random one :+1: